test/unittest/tests/wrappers/connection_wrapper.h

```diff
@@ -37,11 +37,18 @@ class ConnectionWrapper {
     WT_CONNECTION_IMPL *getWtConnectionImpl() const;
     WT_CONNECTION *getWtConnection() const;
 
+    void
+    clearDoCleanup()
```

Comment:
I find the function name and the state mutating semantics of this difficult to follow.
It is probably beyond the scope of this PR but I think setting up the db home dir should be separate from this class: possibly an object that is passed into this class: that way it could take a parameter in its ctor that indicates whether or not the dir should be cleaned up.

Reply:
Hi @marcbutler , I think that further work can be done to improve the wrapper classes, and I think that is outside the scope of this ticket, but well worth doing.

Reply:
agreed